h2o-py/h2o/grid/grid_search.py

```diff
@@ -72,17 +76,20 @@ class H2OGridSearch(h2o_meta(Keyed)):
 
 
     def __init__(self, model, hyper_params, grid_id=None, search_criteria=None, export_checkpoints_dir=None,
-                 parallelism=1):
+                 recovery_dir=None, parallelism=1):
```

Comment:
This API requires the user to specify the dir. What if H2O is managed ? I believe in managed runs, like on Kubernetes, an automatic regime would be useful. How are we going to solve this ? By introducing another parameter in future, e.g. `autorecovery` ? 

Could be solved by enums in Py (not sure how to solve this in R, but S4 classes seems to offer a solution) and introduce two modes for a single `recovery` parameter. Pseudocode:

enum Recovery{
Directory(Path directoryToSaveTo),
Automated
}

The automated mode could simply dump it to a well known location based on system, e.g. /var/h2o on linux.

Reply:
This API is intended for purely manual use case - see the tests that use it, basically the user is responsible for starting a new cluster and loading the grid and continuing training.
For K8s and other auto-recoverable envs I would imagine this would be done somewhat automatically by the back-end. So the user would only say the cluster should auto-recover and all this magic will happen behind the scenes.

Reply:
I see.  Seems fine to me.